### PR TITLE
Skip autoreply for mailing list mails

### PR DESCRIPTION
**Pull request by harobed**
Skip autoreply for mailing list mails (based on https://tools.ietf.org/html/rfc5230#page-7 )

### CODE REVIEW

File: modoboa_postfix_autoreply/management/commands/autoreply.py [hunk 1]
```diff
@@ -85,8 +85,33 @@ def handle(self, *args, **options):
...
+        sender_localpart = sender.lower().split('@')[0]
+        if (
+            (sender_localpart in ('mailer-daemon', 'listserv', 'majordomo')) or
+            (sender_localpart.startswith('owner-')) or
+            (sender_localpart.endswith('-request'))
+        ):
+            logger.debug("Skip auto reply, this mail come from mailing list")
+            sys.exit(0)
+
+        for line in sys.stdin:
```
Comment (tonioo) [inline modoboa_postfix_autoreply/management/commands/autoreply.py:101]:
``` python
line.strip("\n")
if not line:
    break
```

File: modoboa_postfix_autoreply/management/commands/autoreply.py [hunk 2]
```diff
@@ -85,8 +85,33 @@ def handle(self, *args, **options):
...
+            if line == "\n":
+                break
+
+            if (
+                (line == "Precedence: bulk\n") or
+                (line == "X-Mailer: PHPMailer\n") or
+                line.startswith('List-')
+            ):
```
Comment (tonioo) [inline modoboa_postfix_autoreply/management/commands/autoreply.py:110]: Now you can  remove trailing \n

File: modoboa_postfix_autoreply/management/commands/autoreply.py [hunk 3]
```diff
@@ -85,8 +85,33 @@ def handle(self, *args, **options):
...
+                logger.debug(
+                    "Skip auto reply, this mail come from mailing list")
```
Comment (tonioo) [inline modoboa_postfix_autoreply/management/commands/autoreply.py]: this mail comes from a mailing list

File: modoboa_postfix_autoreply/management/commands/autoreply.py [hunk 4]
```diff
@@ -85,8 +85,33 @@ def handle(self, *args, **options):
...
+        # https://tools.ietf.org/html/rfc5230#page-7
         sender = args[0]
+
+        sender_localpart = sender.lower().split('@')[0]
+        if (
+            (sender_localpart in ('mailer-daemon', 'listserv', 'majordomo')) or
+            (sender_localpart.startswith('owner-')) or
+            (sender_localpart.endswith('-request'))
+        ):
+            logger.debug("Skip auto reply, this mail come from mailing list")
```
Comment (tonioo) [inline modoboa_postfix_autoreply/management/commands/autoreply.py]: this mail comes from a mailing list

File: modoboa_postfix_autoreply/management/commands/autoreply.py [hunk 5]
```diff
@@ -85,8 +85,33 @@ def handle(self, *args, **options):
             "autoreply sender=%s recipient=%s", args[0], ",".join(args[1:])
         )
 
-        PostfixAutoreply().load()
+        # Mailing list filter based on
+        # https://tools.ietf.org/html/rfc5230#page-7
         sender = args[0]
+
+        sender_localpart = sender.lower().split('@')[0]
```
Comment (tonioo) [inline modoboa_postfix_autoreply/management/commands/autoreply.py]: You can use the split_mailbox function provided by modoboa.lib.email_utils

**Comment by harobed** [comment]
My code have been updated with your comments.

**Comment by harobed** [comment]
Thanks :)

**Pull request merged**